libr/main/rasm2.c

```diff
@@ -42,6 +45,9 @@ static RAsmState *__as_new() {
 }
 
 static void __as_free(RAsmState *as) {
+	if (as->a) {
```

Comment:
This may cause a double free because asm doesnt owns this instance when used from Core

Reply:
Again, this doesn't happen because of https://github.com/radareorg/radare2/pull/16630#discussion_r411462961.

Reply:
Ah ok this code is for rasm2